cylc/flow/scripts/cylc_ping.py

```diff
@@ -45,22 +47,59 @@ def get_option_parser():
 def main(parser, options, suite, task_id=None):
     pclient = SuiteRuntimeClient(suite, timeout=options.comms_timeout)
 
-    # cylc ping SUITE
-    pclient('ping_suite')  # (no need to check the result)
-    if cylc.flow.flags.verbose:
-        host, port = SuiteRuntimeClient.get_location(suite)
-        sys.stdout.write("Running on %s:%s\n" % (host, port))
-    if task_id is None:
-        sys.exit(0)
-
-    # cylc ping SUITE TASKID
-    if not TaskID.is_valid_id(task_id):
+    if task_id and not TaskID.is_valid_id(task_id):
         raise UserInputError("Invalid task ID: %s" % task_id)
-    success, msg = pclient('ping_task', {'task_id': task_id})
 
-    if not success:
-        print(cparse(f'<red>{msg}</red>'))
-        sys.exit(1)
+    flow_kwargs = {
+        'request_string': '''
+            query ($wFlows: [ID]) {
+              workflows(ids: $wFlows) {
+                id
+                name
+                port
+                pubPort
+              }
+            }
+        ''',
+        'variables': {'wFlows': [suite]}
```

Comment:
Is this using a running workflow to list other running workflows?

Or is the intention to use the UIS?

Reply:
Artifact of UIS use, if the data-store had more than one workflow (every data-store is structured `self.data[w_id]`) or the resolvers used scan to send the query off to other workflows (without causing a network storm! 🌩️ ) we could..

Reply:
Also, yes (if that's what you meant, sorry!); once the CLI goes through the UIS this will kick in to target multiple workflows.

Reply:
> if the data-store had more than one workflow or the resolvers used scan to send the query off to other workflows

???

Reply:
If the UIS is present we can get a workflow listing from there.

If the UIS is not present, how would this work? Surely we would need to do a FS scan to discover workflows?

Reply:
> If the UIS is present we can get a workflow listing from there.

Yes, this is already in place. The UIS mutation resolvers are different from the WFS, they parse the `ids`/`workflows` arguments to figure out which workflow(s) to send it to.

> If the UIS is not present, how would this work? Surely we would need to do a FS scan to discover workflows?

Correct, we would have to use resolvers like those at the UIS to parse the arguments against a list of workflow IDs.. However, we wouldn't want duplicate parsing at both UIS and WFS (so that might have to be an arg too!)

Reply:
You can actually test it out in the WUI GraphiQL, just use a glob for the workflow IDs i.e.:
```
mutation {
  stop(workflows: ["*"]) {
    result
  }
}
```
(off the top of my head)